src/textual/css/_styles_builder.py

```diff
@@ -128,7 +128,7 @@ def _process_space(self, name: str, tokens: list[Token]) -> None:
         append = space.append
         for token in tokens:
             (token_name, value, _, _, location) = token
-            if token_name == "scalar":
+            if token_name in ("number", "scalar"):
```

Comment:
Missed this in my PR yesterday which split out tokenising of number/scalar. It broke margin/padding when you specified any of the values in CSS as a number rather than scalar (i.e. no units).